mailchimp3/entities/landingpages.py

```diff
@@ -0,0 +1,106 @@
+# coding=utf-8
+"""
+The Landing Pages API endpoint
+
+Documentation: https://mailchimp.com/developer/reference/landing-pages/
+Schema: https://api.mailchimp.com/schema/3.0/LandingPages/Instance.json
+"""
+from __future__ import unicode_literals
+
+from mailchimp3.baseapi import BaseApi
+from mailchimp3.entities.landingpageaction import LandingPageAction
+from mailchimp3.entities.landingpagecontent import LandingPageContent
+
+
+class LandingPages(BaseApi):
+    """
+    Manage your Landing Pages, including publishing and unpublishing.
+    """
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the endpoint
+        """
+        super(LandingPages, self).__init__(*args, **kwargs)
+        self.endpoint = 'landing-pages'
+        self.page_id = None
+        self.content = LandingPageContent(self)
+        self.actions = LandingPageAction(self)
+
+
+    def create(self, data):
+        """
+        Create a new Landing Page.
+
+        :param data: The request body parameters
+        :type data: :py:class:`dict`
+        """
+        if 'list_id' not in data:
+            raise KeyError('The landing page must have a list_id')
+
+        response = self._mc_client._post(url=self._build_path(), data=data)
+        if response is not None:
+            self.page_id = response['id']
+        else:
+            self.page_id = None
+        return response
+
+
+    def all(self, get_all=False, **queryparams):
+        """
+        Get all landinge pages.
+
+        :param get_all: Should the query get all results
+        :type get_all: :py:class:`bool`
+        :param queryparams: The query string parameters
+        queryparams['sort_field'] = string
+        queryparams['sort_dir'] = string
+        queryparams['fields'] = []
+        queryparams['exclude_fields'] = []
+        queryparams['count'] = integer
+        """
+        self.page_id = None
+        if get_all:
+            return self._iterate(url=self._build_path(), **queryparams)
+        else:
+            return self._mc_client._get(url=self._build_path(), **queryparams)
+
+
+    def get(self, page_id, **queryparams):
+        """
+        Get information about a specific page.
+
+        :param campaign_id: The unique id for the campaign.
+        :type campaign_id: :py:class:`str`
+        :param queryparams: The query string parameters
+        queryparams['fields'] = []
+        queryparams['exclude_fields'] = []
+        """
+        self.page_id = page_id
+        return self._mc_client._get(url=self._build_path(page_id), **queryparams)
+
+
+    def update(self, page_id, data):
+        """
+        Update a landing page.
+
+        :param page_id: The unique id for the landing page.
+        :type page_id: :py:class:`str`
+        :param data: The request body parameters
+        :type data: :py:class:`dict`
+        """
+        self.page_id = page_id
+        if 'list_id' not in data:
+            raise KeyError('The landing page must have a list_id')
```

Comment:
Hi @stephenross Thanks for the review! Based on my manual testing, an error is raised when `list_id` isn't provided.

```
from mailchimp3 import MailChimp
client = MailChimp(access_token='token')
data = {'name': 'test page', 'title': 'hello', 'description': 'hello world'}
client.landing_pages.create(data)

Traceback
...
MailChimpError: {
'type': 'http://developer.mailchimp.com/documentation/mailchimp/guides/error-glossary/',
'title': 'Invalid Resource', 'status': 400, 'detail': "The resource submitted could not be validated. For field-specific details, see the 'errors' array.",
'instance': '37ad2bcb-582b-4807-ac4c-de09e9a0e90a',
'errors': [{'field': '', 'message': 'Required fields were not provided: list_id'}]
}
```

Reply:
Cool, that works for me, if you have any ability to do so, please notify MailChimp that the documentation is incorrect so that they can update the documentation.

Reply:
Looking at the sample, I see I was mistaken, that is a failure on attempting to create a new landing page (the POST request), it is not a failure on updating a landing page (the PATCH request). Please test with the update method and verify that the failure is on the update and not the create. Looking at the documentation, you should be able to take an existing landing page and send an empty data dict without error.

Reply:
Hi @stephenross, pardon me if I'm wrong but seems the documentation says that it's required with the `*` sign beside `list_id` when doing `POST` request. 

https://mailchimp.com/developer/reference/landing-pages/#post_/landing-pages
<img width="784" alt="image" src="https://user-images.githubusercontent.com/1899227/69143282-8d33a180-0b03-11ea-8050-fac72380c736.png">

You're correct with `PATCH`. It should be able to handle an empty data dict. I have updated the branch to remove the condition.

Reply:
You are correct, list_id with the * shown in the screenshot is the required item in the data for that request. You currently have all of the request body parameters listed in the docstring for create, you can remove all but list_id from the `data = {}` block.

Reply:
Got it. I removed the other items on the dict data.